client/src/components/PacmanGame/PacmanGame.jsx

```diff
@@ -24,35 +28,87 @@ class PacmanGame extends Component {
     entityApplier(getPacman(), entityToCode('pacman'));
-    console.log((gridState[43][5]));
-    // console.log(gridState[43])
     entityApplier(getWalls(cellsInEachRow), entityToCode('wall'));
     entityApplier(getGhosts(), entityToCode('ghost'));
-    console.log((gridState[43][5]));
     entityApplier(getEnergizers(), entityToCode('energizer'));
 
-    this.setState({ 'gridState': gridState });
-    this.setState({ 'gridSize': gridSize });
+    this.setState({ gridState, gridSize });
 
-    console.log(getPacman());
-    console.log((gridState[43][5]));
-    console.log('food', (gridState[15][9]));
+    let walls = [];
+    gridState.forEach((rowArray, rowIndex) => {
+      rowArray.forEach((cell, columnIndex) => {
+        if (cell === 4) {
+          walls.push([rowIndex, columnIndex]);
+        }
+      })
+    });
+    this.setState({ walls });
+  }
+  startGame = () => {
+    this.animationHandler = setInterval(
+      this.animateGame,
+      this.state.config.refreshRate
+    );
+    document.addEventListener('keydown', this.setDirection);
+  }
+  checkCollision = ({ x, y }) => {
+    let walls = this.state.walls;
+    let hasCollision = false;
 
-    // console.log('food', getFoods());
-    console.log('pacman', getPacman());
-    // console.log(getWalls());
-    console.log('ghost', getGhosts());
+    walls.forEach((wall) => {
+      if (wall[0] === x && wall[1] === y) {
+        hasCollision = true;
+      }
+    });
+    return hasCollision;
+  }
+  animateGame = () => {
+    const initialLocation = this.state.pacman;
+    const newLocation = { ...initialLocation };
+
+    const direction = this.state.direction;
+    if (direction === 'RIGHT') {
+      newLocation['x'] = initialLocation['x'] + 1;
+    } else if (direction === 'LEFT') {
+      newLocation['x'] = initialLocation['x'] - 1;
+    } else if (direction === 'UP') {
+      newLocation['y'] = initialLocation['y'] - 1;
+    } else if (direction === 'DOWN') {
+      newLocation['y'] = initialLocation['y'] + 1;
+    }
+    if (!this.checkCollision(newLocation)) {
+      this.setState({
+        pacman: newLocation
+      });
+    }
+  }
+  setDirection = ({ which: keycode }) => {
+    const oldDirection = this.state.direction;
+    let dir;
+    if (keycode === 37 && oldDirection !== 'RIGHT') dir = 'LEFT';
+    if (keycode === 38 && oldDirection !== 'DOWN') dir = 'UP';
+    if (keycode === 39 && oldDirection !== 'LEFT') dir = 'RIGHT';
+    if (keycode === 40 && oldDirection !== 'UP') dir = 'DOWN';
 
+    this.setState({
+      direction: dir
+    });
   }
+  componentDidMount() {
+    this.setInitialGameState();
+  }
+
   render() {
-    return <PacmanBoard gridSize={this.state.gridSize} gridState={this.state.gridState} />
+    return (
+      <div>
+        <button className="" onClick={this.startGame}>Start</button>
+        <PacmanBoard gridSize={this.state.gridSize} gridState={this.state.gridState} pacman={this.state.pacman} />
```

Comment:
Use destructuring.

 @Shalinit3 Why is husky not stopping from committing code that fails style guide?